code highlighting

SachaKor opened the pull request.

porchy13 commented:
The idea of using the Service Provider is really good. It just lacks of a default highlighter that is returned by default by ```Driver```. So, the implementation that have a specific highlighter can return the specific one and the other one will use the default highlighter.

The default highlighter should be based on the SQL 2003 reserved words, those already defined in the ```highlighter``` module.

porchy13 commented:
After reflection, the CSS file should not be handled by an implementation of an ```Highlighter```. The reason is that it will be difficult to change the style of the app.

Let's say we want a dark mode, the highlighting rules should be defined at the level of the app and not at the level of the highlighter. The writing of SQL queries is the job of a developer and not of the implementer of the highlighter. The developer should be confortable in it's writing environnement.

SachaKor commented:
I moved the CSS file handling from the highlighters to the app module. Now it is statically stored there. I also moved the computeHighlighting() method to the EditorController class. Otherwise the developper of the highlighter would still be aware of the CSS syntax to associate the matcher group with it's CSS style class.
